### DISCUSSION TITLE
Sensor Install cannot enroll

### DISCUSSION THREAD

**Discussion by gormatoobay** [Security-Onion-Solutions/securityonion#2665] [Unsupported Versions] [answered]
attempt to reach manager during sensor install / enrollment fails at the sensor node while attempting to copy so-remote ssh key to manager.  Logs indicate the manager was not reached (salt log is clean).

Error output:
soremote@<manager_hostname>: Permission denied (publickey).
rm: cannot remove '/root/install_opt': No such file or directory
Could not determine version of Security Onion running on manager <manager_hostname>. Please check your network settings and run setup again.

I can generate the "soremote@<manager_hostname>: Permission denied (publickey)." error by executing "ssh so-remote@<manager_hostname> omitting the identity file... 

my question - how is security onion going to successfully  enroll without an identity file?  Where was that setup step in the instructions?  I must get my glasses checked.

**Comment by weslambert** [comment]
Was there any point in which network connectivity/firewall could have affected the initial copy?

**Reply by gormatoobay** [reply]
"affected the initial copy" - do you mean the install of the manager, or the initial copy of the ssh key for soremote to the manager?  I do the the initial copy of the ssh key failed, but I don't know what log on the manager to examine, nor do I know what port is needed to perform the copy.  I can confirm ssh is open at all points. I tried the install again on a fresh vm.

**Reply by gormatoobay** [reply]
here is the log on the aborted sensor side, this gives a better picture, but not a full picture regarding key generation and failure to copy.

Notice that it failed to retrieve the correct version, and really got sideways using a <blank> version as part of the tar filename to download.

Generating SSH key
Removing old entry for manager from known_hosts if it exists
Copying the SSH key to the manager
/usr/bin/ssh-copy-id: INFO: Source of key(s) to be installed: "/root/.ssh/so.key.pub"
soremote@<hostname>: Permission denied (publickey).
Could not determine version of Security Onion running on manager <hostname>. Please check your network settings and run setup again.
2021-01-21T04:48:11Z | I | Installer version mismatch, downloading correct version from manager
soremote@<hostname>: Permission denied (publickey).
soremote@<hostname>: Permission denied (publickey).
Could not download .tar.gz from manager, please check your network settings and verify the file /opt/so/repo/.tar.gz exists on the manager.

**Comment by weslambert** [comment, marked as answer]
Is this being installed in AWS, or is this local? It looks like so-remote is now expecting key auth, so you can't connect to the manager via password to copy the key. If it is in AWS, then you'll need to allow password auth in `/etc/ssh/sshd_config`.

**Reply by gormatoobay** [reply]
Yes, these are AWS instances, and your instruction addressed my problem, thanks!
